### PR TITLE
gnrc_sock_ip: fix memcpy()/memset() sizeof-type

### PR DESCRIPTION
**Pull request by miri64**
The type in the `sizeof()` is just plain wrong. This fixes it.

### CODE REVIEW

File: sys/net/gnrc/sock/ip/gnrc_sock_ip.c [hunk 1]
```diff
@@ -41,17 +41,17 @@ int sock_ip_create(sock_ip_t *sock, const sock_ip_ep_t *local,
         if (gnrc_af_not_supported(local->family)) {
             return -EAFNOSUPPORT;
         }
-        memcpy(&sock->local, local, sizeof(sock_ip_t));
+        memcpy(&sock->local, local, sizeof(sock_ip_ep_t));
```
Comment (keestux) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: Take a look at all the memset's and memcpy's. We could use some consistency.

Yes, of course you must know what the variable is. But hey, we're programmers. We know. :-)

Comment (miri64) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: Fixed

File: sys/net/gnrc/sock/ip/gnrc_sock_ip.c [hunk 2]
```diff
@@ -36,7 +36,7 @@ int sock_ip_create(sock_ip_t *sock, const sock_ip_ep_t *local,
         (local->netif != remote->netif)) {
         return -EINVAL;
     }
-    memset(&sock->local, 0, sizeof(sock_ip_t));
+    memset(&sock->local, 0, sizeof(sock_ip_ep_t));
```
Comment (keestux) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: Please consider my suggestion about using a variable instead of a type. This bug wouldn't exist if we did it that way.

Comment (keestux) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: And, hmm, that was a quick fix. It compiled, so it must be correct ? ;-)

File: sys/net/gnrc/sock/ip/gnrc_sock_ip.c (hunk 1 above)
Comment (miri64) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: Can we not discuss minor style questions in a hotfix for the release, please?

File: sys/net/gnrc/sock/ip/gnrc_sock_ip.c (hunk 2 above)
Comment (miri64) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: > Well than I would like to be consistent with gnrc_sock_udp where I also used size notation ;-).

Comment (miri64) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: > And, hmm, that was a quick fix. It compiled, so it must be correct ? ;-)

Nope, tested with `tests/gnrc_sock_ip` on IoT-LAB, but since this and the following variable were initialized with 0 it fell through.

Comment (OlegHahm) [inline sys/net/gnrc/sock/ip/gnrc_sock_ip.c]: I have mixed feelings whether to use the type or the instance as a parameter for `sizeof` here. I understand @keestux's rationale, but I also think that using the type makes the developer aware of how much memory is actually initialized/copied.

However, I totally agree with @miri64 that this is out of scope for this PR.

**Comment by miri64** [comment]
Backport provided in #6098
